golem-component-service/src/api/component.rs

```diff
@@ -188,16 +219,57 @@ impl ComponentApi {
         component_type: Query<Option<ComponentType>>,
     ) -> Result<Json<Component>> {
         let record = recorded_http_api_request!(
-            "update_component",
+            "upload_component",
             component_id = component_id.0.to_string()
         );
+
         let response = {
             let data = wasm.0.into_vec().await?;
             self.component_service
                 .update(
                     &component_id.0,
                     data,
                     component_type.0,
+                    None,
+                    &DefaultNamespace::default(),
+                )
+                .instrument(record.span.clone())
+                .await
+                .map_err(|e| e.into())
+                .map(|response| Json(response.into()))
+        };
+        record.result(response)
+    }
+
+    /// Update a component
+    #[oai(
+        path = "/:component_id/updates",
+        method = "post",
```

Comment:
Update (the existing one too) is not idempotent, so imo post is a better fit then push.

This old one is left for backwards compat, as we now need to be multipart here too.

Reply:
May be you are right, but I am having my doubts :) 
So let me confirm, the existing `/:component_id/upload` with `PUT` doesn't work here?
I guess update should be merging with the previous files if there are no conflicts.

Reply:
@afsalthaj the IFS is tied for a specific component version. No merging.